[PLAT-1854] Document dynamic resource labels for Cloud (#739)
seqeralabs/docs: platform-cloud/docs/resource-labels/overview.md

tags: [resource labels, labels]
---

Platform supports resource labels in runs, compute environments, pipelines, and actions. This offers a flexible tagging system for annotation and tracking of the cloud services consumed by a run. Resource labels are sent to the cloud service provider in `key=value` format.

Resource labels are applied to elements during:

- Compute environment creation
- Workflow submission
- Workflow execution

### Dynamic resource labels 

> Use dynamic resource labels to tag cloud resources with Platform run and Nextflow session identifiers at workflow submission and execution time. By providing a variable value (either `${workflowId}` or `${sessionId}`) in the standard `key=value` resource label format, the unique run or session ID for each pipeline run will be propagated to your cloud provider for all the resources spawned by that particular run. 

Reviewer: ```suggestion
Use dynamic resource labels to tag cloud resources with Platform run and Nextflow session identifiers at workflow submission and execution time. When you provide a variable value (either `${workflowId}` or `${sessionId}`) in the standard `key=value` resource label format, the unique run or session ID for each pipeline run will be propagated to your cloud provider for all the resources spawned by that particular run.
```

> For example, for user Alex to apply labels containing the unique Platform run ID to all the resources spawned by a run, they could apply a dynamic label such as `alex=run-${workflowId}` to their compute environment, pipeline, or manually in the pipeline launch form. 

Reviewer: ```suggestion
For example, to apply labels containing the unique Platform run ID to all the resources spawned by a run, add a dynamic label such as `alex=run-${workflowId}` to your compute environment, pipeline, or manually in the pipeline launch form. 
```

Search and filter pipelines and runs using one or more resource labels. The resource label search uses a `label:key=value` format.

### Manage workspace resource labels

Select a workspace's **Settings** tab to view all the resource labels used in that workspace. Resource labels can only be edited or deleted by admins and only if they're not already associated with **any** resource. This applies to resource labels associated with compute environments and runs. When you add or edit a resource label, you can optionally set the **Use as default in compute environment form** option. Workspace default resource labels are prefilled in the **Resource labels** field when creating a new compute environment in that workspace.

The deletion of a resource label from a workspace has no influence on the cloud environment.

### Resource label propagation to cloud environments

:::note
- Cloud provider credentials must have the appropriate roles or permissions to tag resources in your environment.

> - You can't assign multiple resource labels, using the same key, to the same resource — regardless of whether this option is supported by the destination cloud provider.

Reviewer: ```suggestion
- You can't assign multiple resource labels, using the same key, to the same resource, regardless of whether this option is supported by the destination cloud provider.
```
